app/controllers/api/submission_files_controller.rb

```diff
@@ -199,10 +163,91 @@ def remove_folder
       end
     end
 
+    def submit_file
+      student = Student.find_by(user: @real_user, course: @current_course)
```

Comment:
We already have the `current_role` method that computes this, so you can do `student = current_role`